### PR TITLE
tsimp-- swc++

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
…for checking

# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [x] Bug Fix

## Linked tickets

VR-222

## High level description

Replaces `tsimp` with `swc`

## Detailed description

`tsimp` seems dead and therefore we're getting ahead and trying a switch to `@swc-node/register. Note as this changes our dev compiler it also makes sense to switch to `swc` for all compilation for consistency. Note this is a pretty substantial change conceptually. 

Some of the points to note:

* `swc` does not do any type checking so we are still using `tsc` for that as recommended. We must therefore use `npm run check` in our workflows as compiled code could contain invalid types
* The `swc` config needs to be kept in line with the `tsc` config. Only time will tell how easy this is
* I've fixed the `watch` workflow for tsoa which then generated an infinite loop of nodejs compiles. We can fix this for the `routes.ts` file by telling `tsoa` not to write to disk if this is unchanged but no similar option is avalable for the swagger file. I've instead changed config to move the `swagger.json` to the `build` directory as it is a fully built file. This has made the importing of the swagger.json more interesting but the benefit is everything still ends up in `build` in prod
* Note I have tested out using vs-code developer tools. source maps seem to be working fine and dev ex seems good

## Describe alternatives you've considered

#174 . Would lead to a much wider change and worse dev ex. 

## Operational impact

None

## Additional context

None

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "veritable-ui",
-  "version": "0.8.73",
+  "version": "0.8.74",
```
Comment (dblane-digicatapult) [inline package.json:3]:
```suggestion
  "version": "0.9.0",
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: src/swagger.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export default async function loadApiSpec(env: Env): Promise<unknown> {
   const authorizationUrl = `${env.get('IDP_PUBLIC_URL_PREFIX')}${env.get('IDP_AUTH_PATH')}`
   const tokenUrl = `${env.get('IDP_PUBLIC_URL_PREFIX')}${env.get('IDP_TOKEN_PATH')}`
 
-  const swaggerBuffer = await fs.readFile(path.join(__dirname, './swagger.json'))
+  const swaggerBuffer = await fs.readFile(path.join(__dirname, '..', 'build', 'swagger.json'))
```
[the remark this answers is not in the document]
Comment (mattdean-digicatapult) [reply src/swagger.ts:20]: Generally speaking you shouldn't construct paths yourself as different OS use different separators. `/` is a unix path separator but `\` is used on windows. I know we don't run on windows but it's good practice

File: package.json (hunk 1 above)
Comment (mattdean-digicatapult) [reply package.json:3]: Switching to minor using the bot

**Review by dblane-digicatapult — APPROVED** [review]

**Review by n3op2 — APPROVED** [review]

File: src/swagger.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (n3op2) [reply src/swagger.ts:20]: makes sense, thanks

**Pull request merged**